components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/AccessTokenIssuer.java

```diff
@@ -205,15 +205,19 @@ public OAuth2AccessTokenRespDTO issue(OAuth2AccessTokenReqDTO tokenReqDTO)
             AuthorizationGrantCacheKey cacheKey = new AuthorizationGrantCacheKey(getAuthorizationCode(tokenReqDTO));
             AuthorizationGrantCacheEntry authorizationGrantCacheEntry =
                     AuthorizationGrantCache.getInstance().getValueFromCacheByCode(cacheKey);
-            if (authorizationGrantCacheEntry != null &&
-                    authorizationGrantCacheEntry.getAccessTokenExtensionDO() != null) {
-                if (authorizationGrantCacheEntry.getAccessTokenExtensionDO().getRefreshTokenValidityPeriod() >
-                        EXTENDED_REFRESH_TOKEN_DEFAULT_TIME) {
-                    tokReqMsgCtx.setRefreshTokenvalidityPeriod(
-                            authorizationGrantCacheEntry.getAccessTokenExtensionDO().getRefreshTokenValidityPeriod());
+            if (authorizationGrantCacheEntry != null) {
+                if (authorizationGrantCacheEntry.getAccessTokenExtensionDO() != null) {
+                    if (authorizationGrantCacheEntry.getAccessTokenExtensionDO().getRefreshTokenValidityPeriod() >
+                            EXTENDED_REFRESH_TOKEN_DEFAULT_TIME) {
+                        tokReqMsgCtx.setRefreshTokenvalidityPeriod(
+                                authorizationGrantCacheEntry.getAccessTokenExtensionDO()
+                                        .getRefreshTokenValidityPeriod());
+                    }
+                    tokReqMsgCtx.getOauth2AccessTokenReqDTO().setAccessTokenExtendedAttributes(
+                            authorizationGrantCacheEntry.getAccessTokenExtensionDO());
                 }
-                tokReqMsgCtx.getOauth2AccessTokenReqDTO().setAccessTokenExtendedAttributes(
-                        authorizationGrantCacheEntry.getAccessTokenExtensionDO());
+                tokReqMsgCtx.setSelectedAcr(authorizationGrantCacheEntry.getSelectedAcrValue());
```

Comment:
We don't need these values for all the access tokens. Can you add these conditionally only when requested.